state/state.go

```diff
@@ -607,7 +607,8 @@ func (state *HelmState) FilterReleases(labels []string) error {
 		filteredReleases = append(filteredReleases, r)
 	}
 	if len(filteredReleases) == 0 {
-		return fmt.Errorf("specified selector did not match any releases in %s\n", state.file)
+		state.logger.Debugf("specified selector did not match any releases in %s\n", state.file)
+		return nil
```

Comment:
This broke helmfile's ability to filter the unmatched releases. Returning `nil` here means that you filter nothing.

#225 includes the fix for it.